include/cantera/thermo/EEDFTwoTermApproximation.h

```diff
@@ -0,0 +1,262 @@
+     * input from the compute function. Currently, this means that the 2
+     * PlasmaPhases have to have consist of the same species and elements.
+     */
+    PlasmaPhase* m_phase;
+
+    //! Iterate f0 (EEDF) until convergence
+    void converge(Eigen::VectorXd& f0);
+
+    //! An iteration of solving electron energy distribution function
+    Eigen::VectorXd iterate(const Eigen::VectorXd& f0, double delta);
+
+    //! The integral in [a, b] of \f$x u(x) \exp[g (x_0 - x)]\f$
+    //! assuming that u is linear with u(a) = u0 and u(b) = u1
+    double integralPQ(double a, double b, double u0, double u1,
+                       double g, double x0);
+
+    //! Vector g is used by matrix_P() and matrix_Q().
+    /**
+     * \f[
+     * g_i = \frac{1}{\epsilon_{i+1} - \epsilon_{i-1}} \ln(\frac{F_{0, i+1}}{F_{0, i-1}})
+     * \f]
+     */
+    vector<double> vector_g(const Eigen::VectorXd& f0);
+
+    //! The matrix of scattering-out.
+    /**
+     * \f[
+     * P_{i,k} = \gamma \int_{\epsilon_i - 1/2}^{\epsilon_i + 1/2}
+     * \epsilon \sigma_k exp[(\epsilon_i - \epsilon)g_i] d \epsilon
+     * \f]
+     */
+    Eigen::SparseMatrix<double> matrix_P(const vector<double>& g, size_t k);
+
+    //! The matrix of scattering-in
+    /**
+     * \f[
+     * Q_{i,j,k} = \gamma \int_{\epsilon_1}^{\epsilon_2}
+     * \epsilon \sigma_k exp[(\epsilon_j - \epsilon)g_j] d \epsilon
+     * \f]
+     */
+    //! where the interval \f$[\epsilon_1, \epsilon_2]\f$ is the overlap of cell j,
+    //! and cell i shifted by the threshold energy:
+    /**
+     * \f[
+     * \epsilon_1 = \min(\max(\epsilon_{i-1/2}+u_k, \epsilon_{j-1/2}),\epsilon_{j+1/2}),
+     * \f]
+     * \f[
+     * \epsilon_2 = \min(\max(\epsilon_{i+1/2}+u_k, \epsilon_{j-1/2}),\epsilon_{j+1/2})
+     * \f]
+     */
+    Eigen::SparseMatrix<double> matrix_Q(const vector<double>& g, size_t k);
+
+    //! Matrix A (Ax = b) of the equation of EEDF, which is discretized by the exponential scheme
+    //! of Scharfetter and Gummel,
+    /**
+     * \f[
+     *     \left[ \tilde{W} F_0 - \tilde{D} \frac{d F_0}{\epsilon} \right]_{i+1/2} =
+     *     \frac{\tilde{W}_{i+1/2} F_{0,i}}{1 - \exp[-z_{i+1/2}]} +
+     *     \frac{\tilde{W}_{i+1/2} F_{0,i+1}}{1 - \exp[z_{i+1/2}]}
+     * \f]
+     * where \f$ z_{i+1/2} = \tilde{w}_{i+1/2} / \tilde{D}_{i+1/2} \f$ (Peclet number).
+     */
+    Eigen::SparseMatrix<double> matrix_A(const Eigen::VectorXd& f0);
+
+    //! Reduced net production frequency. Equation (10) of ref. [1]
+    //! divided by N.
+    //! @param f0 EEDF
+    double netProductionFreq(const Eigen::VectorXd& f0);
+
+    //! Diffusivity
+    double electronDiffusivity(const Eigen::VectorXd& f0);
+
+    //! Mobility
+    double electronMobility(const Eigen::VectorXd& f0);
+
+    void initSpeciesIndexCS();
+
+    void checkSpeciesNoCrossSection();
+
+    void updateCS();
+
+    void update_mole_fractions();
+
+    void calculateTotalElasticCrossSection();
+
+    void calculateTotalCrossSection();
```

Comment:
These methods need docstrings. Also worth making the names follow the camelCase convention that we follow in the C++ code and being consistent about abbreviations/initialisms (generally preferring to avoid them in method names).